extensions/wikia/ARecoveryEngine/ARecoveryModule.class.php

```diff
@@ -18,16 +15,31 @@ public function isPageFairRecoveryEnabled() {
 	/**
 	 * Checks whether SourcePoint recovery is enabled (on current wiki)
 	 *
-	 * $wgAdDriverEnableSourcePointRecovery === false; // disabled on wiki
-	 * $wgAdDriverEnableSourcePointRecovery === true; // enabled on wiki
+	 * @return bool
+	 */
+	public static function isSourcePointRecoveryEnabled() {
+		global $wgUser, $wgAdDriverEnableSourcePointRecovery;
+
+		return $wgUser->isAnon() && F::app()->checkSkin( [ 'oasis' ] ) && $wgAdDriverEnableSourcePointRecovery;
```

Comment:
so `isRecoveryEnabled()` would check skin and login status? I don't think it's needed. Also, we may want to differ this approach in the future (recover also on mercury?) but it's FFU so not a strong argument ;)